apps/src/code-studio/components/progress/SelectedStudentInfo.jsx

```diff
@@ -62,67 +82,107 @@ export class SelectedStudentInfo extends React.Component {
+        <RadiumFontAwesome
+          icon="caret-left"
+          onClick={this.previousStudent}
+          style={styles.arrow}
+        />
+        <div style={styles.studentInfo}>
+          <div style={styles.name}>{selectedStudent.name}</div>
+          {level.paired && (
             <div>
-              {level.status !== LevelStatus.not_tried
-                ? new Date(level.updated_at).toLocaleString()
-                : i18n.notApplicable()}
+              <div>{i18n.workedWith()}</div>
+              {level.navigator && (
+                <div key={level.navigator}>
+                  {i18n.partner({partner: level.navigator})}
+                </div>
+              )}
+              {level.driver && (
+                <div key={level.driver}>
+                  {i18n.loggedIn({partner: level.driver})}
+                </div>
+              )}
             </div>
+          )}
+          <div style={styles.bubble}>
+            <TeacherPanelProgressBubble level={level} />
           </div>
-        )}
-        {level.submitLevel && (
-          <div>
-            <div style={styles.timeHeader}>{i18n.submittedOn()}</div>
+          {!level.submitLevel && (
             <div>
-              {level.status === LevelStatus.submitted
-                ? new Date(level.updated_at).toLocaleString()
-                : i18n.notApplicable()}
+              <div style={styles.timeHeader}>{i18n.lastUpdatedNoTime()}</div>
+              <div>
+                {level.status !== LevelStatus.not_tried
+                  ? new Date(level.updated_at).toLocaleString()
+                  : i18n.notApplicable()}
+              </div>
             </div>
-            <Button
-              text={i18n.unsubmit()}
-              color="blue"
-              onClick={this.onUnsubmit}
-              id="unsubmit-button-uitest"
-              disabled={level.status !== LevelStatus.submitted}
-            />
-          </div>
-        )}
-        {level.contained && (
-          <div>
-            <Button
-              text={i18n.clearResponse()}
-              color="blue"
-              onClick={this.onClearResponse}
-              disabled={level.status === LevelStatus.not_tried}
-            />
-          </div>
-        )}
+          )}
+          {level.submitLevel && (
+            <div>
+              <div style={styles.timeHeader}>{i18n.submittedOn()}</div>
+              <div>
+                {level.status === LevelStatus.submitted
+                  ? new Date(level.updated_at).toLocaleString()
+                  : i18n.notApplicable()}
+              </div>
+              <Button
+                text={i18n.unsubmit()}
+                color="blue"
+                onClick={this.onUnsubmit}
+                id="unsubmit-button-uitest"
+                disabled={level.status !== LevelStatus.submitted}
+              />
+            </div>
+          )}
+          {level.contained && (
+            <div>
```

Comment:
nit: remove extra div (unless it's needed for a specific reason)

Reply:
the inner conditionals that determine showing the navigator and driver info don't like it if you remove the outer div